### PR TITLE
Remove workflow failure check in favour of Argo-driven version

### PR DESCRIPTION
**Pull request by lieut-data**
#### Summary
Remove the workflow failure check in favour of the new Argo-driven version under development.

#### Ticket Link
None

#### Release Note
```release-note
NONE
```

### CODE REVIEW

File: .github/workflows/server-ci-master.yml [hunk 1]
```diff
@@ -5,9 +5,27 @@ on:
...
-      - mono-repo*
 
 jobs:
   master-ci:
     uses: ./.github/workflows/server-ci-template.yml
     secrets: inherit
+  on-failure:
+    runs-on: ubuntu-latest
+    needs: master-ci
+    if: ${{ always() && needs.master-ci.result == 'failure' }}
```
Comment (mvitale1989) [inline .github/workflows/server-ci-master.yml:16]: TIL that `always() && needs.master-ci.result == 'failure'` is not equivalent to just `needs.master-ci.result == 'failure'` :open_mouth:  That's very useful to know, thanks Jesse!

File: .github/workflows/server-ci-master.yml [hunk 2]
```diff
@@ -5,9 +5,27 @@ on:
...
+    steps:
+      - name: Report failure
+        env:
+          REPOSITORY: ${{ github.repository }}
+          HEAD_BRANCH: ${{ github.event.workflow_run.head_branch }}
```
Comment (mvitale1989) [inline .github/workflows/server-ci-master.yml:21]: This wouldn't work. The [`github.event.workflow_run.*` event data](https://docs.github.com/en/webhooks/webhook-events-and-payloads#workflow_run) is only available in a `workflow_run`-triggered workflow, but after this change the context for this workflow is within a `push` event.
Let me check the docs for the correct equivalent interpolation

**Review by toninis — CHANGES_REQUESTED** [review]
@lieut-data We can keep the same logic . No need to add this to the rest workflows

File: .github/workflows/server-ci-master.yml (hunk 2 above)
Comment (mvitale1989) [reply .github/workflows/server-ci-master.yml:21]: Ah true, as [Antonis mentions](https://github.com/mattermost/mattermost/pull/25366#pullrequestreview-1720337002) the remediation of simply putting the untrusted interpolations within intermediary variables should be all we need, according to the Jira ticket.
I think we can still keep the failure report within its `workflow_run` file, in this case

File: .github/workflows/server-ci-master.yml [hunk 3]
```diff
@@ -5,9 +5,27 @@ on:
...
+          SERVER_URL: ${{ github.server_url }}
+          WORKFLOW_RUN_ID: ${{ github.event.workflow_run.id}}
+          WORKFLOW_RUN_NAME: ${{ github.event.workflow_run.name }}
+        run: |
+          curl \
+            --fail \
+            -X POST \
+            -H "Content-Type: application/json" \
+            -d "{\"text\":\"#### ⚠️  $REPOSITORY/$HEAD_BRANCH - $WORKFLOW_RUN_NAME build failure ⚠️\\nThe build is failing: [view failure]($SERVER_URL/$REPOSITORY/actions/runs/$WORKFLOW_RUN_ID).\\n\"}" \
+            ${{ secrets.MM_COMMUNITY_DEVELOPERS_INCOMING_WEBHOOK_FROM_GH_ACTIONS }}
```
Comment (mvitale1989) [inline .github/workflows/server-ci-master.yml:31]: I wonder if this variable can be accessed by untrusted workflows? Let me double check, but perhaps this was the reason we needed to have this workflow as privileged :thinking:

Comment (mvitale1989) [reply .github/workflows/server-ci-master.yml:31]: Never mind, afaiu this should run just fine, as long as it doesn't run within repo forks.

**Review by toninis — APPROVED** [review]

**Pull request merged**